tests/executables_src/testHierarchyModifyingGroup.cc

```diff
@@ -0,0 +1,234 @@
+/*
+ * testHierarchyModifyingGroup.cc
+ */
+
+#include <chrono>
+#include <future>
+
+#define BOOST_TEST_MODULE testHierarchyModifyingGroup
+
+#include <boost/mpl/list.hpp>
+#include <boost/test/included/unit_test.hpp>
+#include <boost/thread.hpp>
+
+#include "ApplicationCore.h"
+
+using namespace boost::unit_test_framework;
+namespace ctk = ChimeraTK;
+
+/*********************************************************************************************************************/
+
+struct TestGroup : public ctk::HierarchyModifyingGroup {
+  using ctk::HierarchyModifyingGroup::HierarchyModifyingGroup;
+  ctk::ScalarPushInput<int> myVar{this, "myVar", "MV/m", "Descrption"};
+};
+
+struct TestApplication : public ctk::Application {
+  TestApplication() : Application("testSuite") {}
+  ~TestApplication() { shutdown(); }
+
+  void defineConnections() override {}
+
+  struct TestModule : ctk::ApplicationModule {
+    using ctk::ApplicationModule::ApplicationModule;
+
+    TestGroup a{this, "VariableGroupLike", "Use like normal VariableGroup", {"TagA"}};
+    TestGroup b{this, "/MoveToRoot", "Use like normal VariableGroup with MoveToRoot", {"TagB"}};
+    TestGroup c{this, "../oneUp", "Use like normal VariableGroup with oneUp", {"TagC"}};
+    TestGroup d{this, "..", "Use like normal VariableGroup with oneUpAndHide", {"TagD"}};
+    TestGroup e{this, "local/hierarchy", "Create hierarchy locally", {"TagE"}};
+    TestGroup f{this, "/AtRoot/hierarchy", "Create hierarchy at root", {"TagF"}};
+    TestGroup g{this, "../oneUp/hierarchy", "Create hierarchy one level up", {"TagG"}};
+    TestGroup h{this, "local/very/deep/hierarchy", "Create deep hierarchy locally", {"TagH"}};
+    TestGroup i{this, "/root/very/deep/hierarchy", "Create deep hierarchy at root", {"TagI"}};
+    TestGroup j{this, "../oneUp/very/deep/hierarchy", "Create deep hierarchy one level up", {"TagJ"}};
+    TestGroup k{this, "//extra//slashes////everywhere///", "Extra slashes", {"TagK"}};
```

Comment:
What is the expected behavior of having ".." within absolute paths - Like the somewhat useless "/A/B/../C" (probably coverd by the bad paths test) and the not-so-obvious "/A/B/../../../C" ... etc. Shouldn't we test that as well?

Reply:
"/A/B/../C" - since every layer is handled separately, the test would not test anything else than the test `m` with "hierarchy/with/../dots/../../anywhere/..".

The test for "/A/B/../../../C" is missing though (it should throw an exception). I will add this.